.eslintrc.js

```diff
@@ -0,0 +1,215 @@
+    ],
+    'rules': {
+      // TS rules
+      // Blockly uses namespaces to do declaration merging in some cases.
+      '@typescript-eslint/no-namespace': ['off'],
+      // Use the updated TypeScript-specific rule.
+      'no-invalid-this': ['off'],
+      '@typescript-eslint/no-invalid-this': ['error'],
+      // Needs decision. 601 problems.
+      '@typescript-eslint/no-non-null-assertion': ['off'],
+      // Use TS-specific rule.
+      'no-unused-vars': ['off'],
+      '@typescript-eslint/no-unused-vars': [
+        'warn', {
+          'argsIgnorePattern': '^_',
+          'varsIgnorePattern': '^_',
+        }
+      ],
+      'func-call-spacing': ['off'],
+      '@typescript-eslint/func-call-spacing': ['warn'],
+      // Temporarily disable. 23 problems.
+      '@typescript-eslint/no-explicit-any': ['off'],
+      // Temporarily disable. 128 problems.
+      'require-jsdoc': ['off'],
+      // Temporarily disable. 55 problems.
+      '@typescript-eslint/ban-types': ['off'],
+      // Temporarily disable. 33 problems.
+      '@typescript-eslint/no-empty-function': ['off'],
+      // Temporarily disable. 3 problems.
+      '@typescript-eslint/no-empty-interface': ['off'],
+
+      // TsDoc rules (using JsDoc plugin)
+      // Disable built-in jsdoc verifier.
+      'valid-jsdoc': ['off'],
+      // Don't require types in params and returns docs.
+      'jsdoc/require-param-type': ['off'],
+      'jsdoc/require-returns-type': ['off'],
+      // params and returns docs are optional.
+      'jsdoc/require-param-description': ['off'],
+      'jsdoc/require-returns': ['off'],
+      // Disable for now (breaks on `this` which is not really a param).
+      'jsdoc/require-param': ['off'],
+      // Don't auto-add missing jsdoc. Only required on exported items.
+      'jsdoc/require-jsdoc': [
+        'warn',
+        {
+          'enableFixer': false,
+          'publicOnly': true,
+        },
+      ],
+      // Disable because of false alarms with Closure-supported tags.
+      // Re-enable after Closure is removed.
+      'jsdoc/check-tag-names': ['off'],
+      // Re-enable after Closure is removed. There shouldn't even be
+      // types in the TsDoc.
+      // These are "types" because of Closure's @suppress {warningName}
+      'jsdoc/no-undefined-types': ['off'],
+      'jsdoc/valid-types': ['off'],
+      // Disabled due to not handling `this`. If re-enabled,
+      // checkDestructured option
+      // should be left as false.
+      'jsdoc/check-param-names': ['off', {'checkDestructured': false}],
+      // Allow any text in the license tag. Other checks are not relevant.
+      'jsdoc/check-values': ['off'],
+    },
+  };
+}
+
+module.exports = {
+  'rules' : rules,
+  'env' : {
+    'es2020' : true,
+    'browser' : true,
+  },
+  'globals' : {
+    'goog' : true,
+    'exports' : true,
+  },
+  'extends' :
+            [
+              'eslint:recommended',
+              'google',
+            ],
```

Comment:
Go home formatter, you're drunk:

```suggestion
  extends: [
    'eslint:recommended',
    'google',
  ],
```
(and similarly for `overrides`.)